Buildings/Controls/OBC/FDE/DOAS/SupplyFanController.mo

```diff
@@ -0,0 +1,224 @@
+   final quantity="PressureDifference")=500
+    "Maximum down duct static pressure reset value"
+      annotation (Dialog(group="DDSP range"));
+  parameter Real cvDDSPset(
+   min=0,
+   final unit="Pa",
+   final quantity="PressureDifference")=250
+    "Constant volume down duct static pressure set point";
+ parameter Real damSet(
+   min=0,
+   max=1,
+   final unit="1")=0.9
+   "DDSP terminal damper percent open set point";
+ parameter Boolean vvUnit = true
+   "Set true when unit serves variable volume system.";
+
+  // ---inputs---
+  Buildings.Controls.OBC.CDL.Interfaces.BooleanInput occ
+    "True when occupied mode is active"
+    annotation (Placement(transformation(extent={{-142,60},{-102,100}}),
+       iconTransformation(extent={{-140,50},{-100,90}})));
+  Buildings.Controls.OBC.CDL.Interfaces.RealInput mostOpenDam
+    "Most open damper position from all terminal units served."
+      annotation (Placement(transformation(extent={{-142,-24},{-102,16}}),
+        iconTransformation(extent={{-140,14},{-100,54}})));
+  Buildings.Controls.OBC.CDL.Interfaces.BooleanInput supFanStatus
+    "True when supply fan is proven on."
+      annotation (Placement(transformation(extent={{-142,22},{-102,62}}),
+        iconTransformation(extent={{-140,-56},{-100,-16}})));
+  Buildings.Controls.OBC.CDL.Interfaces.RealInput DDSP
+    "Down duct static pressure measurement."
+      annotation (Placement(transformation(extent={{-142,-106},{-102,-66}}),
+        iconTransformation(extent={{-140,-92},{-100,-52}})));
+
+  // ---outputs---
+  Buildings.Controls.OBC.CDL.Interfaces.BooleanOutput supFanStart
+   "Command supply fan to start when true."
+     annotation (Placement(transformation(extent={{102,60},{142,100}}),
+       iconTransformation(extent={{100,32},{140,72}})));
+  Buildings.Controls.OBC.CDL.Interfaces.BooleanOutput supFanProof
+    "True when supply fan is proven on"
+      annotation (Placement(transformation(extent={{102,22},{142,62}}),
+        iconTransformation(extent={{100,-20},{140,20}})));
+  Buildings.Controls.OBC.CDL.Interfaces.RealOutput supFanSpeed
+    "Supply fan speed command"
+      annotation (Placement(transformation(extent={{102,-50},{142,-10}}),
+        iconTransformation(extent={{100,-64},{140,-24}})));
+
+  Buildings.Controls.OBC.CDL.Continuous.LimPID DDSPreset(
+    k=0.0000001,
+    Ti=0.00025,
+    yMax=maxDDSPset,
+    yMin=minDDSPset,
+    reverseAction=true)
+    "Calculates DDSP reset value for VV units."
+      annotation (Placement(transformation(extent={{-72,4},{-52,24}})));
+  Buildings.Controls.OBC.CDL.Continuous.Sources.Constant dampSet(
+    k=damSet)
+    "Most open damper position set point."
+      annotation (Placement(transformation(extent={{-98,4},{-78,24}})));
+  Buildings.Controls.OBC.CDL.Continuous.LimPID conPID(k=0.0000001, Ti=0.00025)
+      annotation (Placement(transformation(extent={{48,-40},{68,-20}})));
+  Buildings.Controls.OBC.CDL.Logical.Sources.Constant conVVunit(
```

Comment:
do not need this instance and `CVorVV` if we can make following instances to be conditional:
- `mostOpenDam` (`if vvUnit`),
- `dampSet` (`if vvUnit`),
- `DDSPreset` (`if vvUnit`),
- `cvDDSPsetpt` (`if not vvUnit`).

Connect both `DDSPreset.y` and `cvDDSPsetpt.y` to `swi.u1`.